mindmeld/models/nn_utils/sequence_classification.py

```diff
@@ -0,0 +1,394 @@
+                msg = f"{self.name} can be best used with '{k}' equal to '{v}'. Other " \
+                      f"values passed through config params might lead to unexpected results."
+                logger.warning(msg)
+            else:
+                params.update({k: v})
+
+        params.update({"embedder_type": "bert"})
+
+        super().fit(examples, labels, **params)
+
+    def _create_optimizer_and_scheduler(self, num_training_steps):
+
+        num_warmup_steps = min(int(0.1 * num_training_steps), self.params.num_warmup_steps)
+        self.params.update({"num_warmup_steps": num_warmup_steps})
+
+        # https://github.com/huggingface/transformers/blob/master/src/transformers/optimization.py
+        # refer `get_linear_schedule_with_warmup` method
+        def lr_lambda(current_step: int):
+            if current_step < num_warmup_steps:
+                return float(current_step) / float(max(1, num_warmup_steps))
+            return max(
+                0.0, float(num_training_steps - current_step) / float(
+                    max(1, num_training_steps - num_warmup_steps))
+            )
+
+        # load a torch optimizer
+        optimizer = self._create_optimizer()
+        # load a lr scheduler
+        scheduler = getattr(torch.optim.lr_scheduler, "LambdaLR")(optimizer, lr_lambda)
+        return optimizer, scheduler
+
+    def _create_optimizer(self):
+        params = list(self.named_parameters())
+        no_decay = ["bias", 'LayerNorm.bias', "LayerNorm.weight",
+                    'layer_norm.bias', 'layer_norm.weight']
+        optimizer_grouped_parameters = [
+            {'params': [p for n, p in params if not any(nd in n for nd in no_decay)],
+             'weight_decay': 0.01},
+            {'params': [p for n, p in params if any(nd in n for nd in no_decay)],
+             'weight_decay': 0.0}
+        ]
+        optimizer = getattr(torch.optim, self.params.optimizer)(
+            optimizer_grouped_parameters,
+            lr=self.params.learning_rate,
+            eps=1e-08,
+            weight_decay=0.01
+        )
+        return optimizer
+
+    def _init_core(self):
+
+        self.bert_model = HuggingfaceTransformersContainer(
+            self.params.pretrained_model_name_or_path,
+            cache_lookup=False
+        ).get_transformer_model()
+        self.dropout = nn.Dropout(
+            p=1 - self.params.embedder_output_keep_prob
+        )
+        if self.params.embedder_output_pooling_type != "first":
+            self.emb_layer_pooling = PoolingLayer(
+                self.params.embedder_output_pooling_type
+            )
+        self.out_dim = self.params.emb_dim
+
+    def _forward_core(self, batch_data):
+
+        bert_outputs = self.bert_model(**batch_data["hgf_encodings"], return_dict=True)
+
+        if self.params.embedder_output_pooling_type != "first":
+            last_hidden_state = bert_outputs.get("last_hidden_state")  # [BS, SEQ_LEN, EMD_DIM]
```

Comment:
Can you add a comment here that `last_hidden_state` actually means the output of the last layer of the model?